processing/src/main/java/io/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -159,8 +159,18 @@ public GroupByQueryQueryToolChest(
 
       final Sequence<Row> subqueryResult = mergeGroupByResults(subquery, runner, context);
       final List<AggregatorFactory> aggs = Lists.newArrayList();
+
       for (AggregatorFactory aggregatorFactory : query.getAggregatorSpecs()) {
-        aggs.addAll(aggregatorFactory.getRequiredColumns());
```

Comment:
@himanshug I tested it out with two historicals, one serving two segments and the other serving three without any replication between them, ran a query that hit both historicals, and didn't see any issues with the results as far as I can tell.

Having said that, I did look into fixing the issue a different way and posted the branch here: https://github.com/dclim/druid/commit/c4ee30286f327f894ca00139041366cf65c1d091
There's still a weird hack on line 187 of GroupByQueryQueryToolChest.java that I had to do to get outer queries on inner post aggregations to work right, but it doesn't seem to affect the results either. If the other implementation seems better, I'll create another PR with it. Thanks for the feedback!

For reference, here's the dataset, super contrived query, and results. I got the same results for both implementations:

Dataset
```
Timestamp				page	language	added
2013-08-31T01:02:33Z	Page1	en			10
2013-08-31T01:02:34Z	Page2	en			20
2013-08-31T01:03:35Z	Page1	en			0
2013-08-31T01:03:36Z	Page3	ja			11
2013-08-31T01:04:37Z	Page3	en			1335
2013-08-31T01:04:38Z	Page3	en			100
2013-08-31T01:08:38Z	Page2	en			0
2013-08-31T01:08:39Z	Page4	fr			560
2013-08-31T01:09:40Z	Page1	fr			123
2013-08-31T01:09:41Z	Page1	en			1337
```
Query
```
{
  "queryType": "groupBy",
  "dataSource": {
    "type": "query",
    "query": {
      "queryType": "groupBy",
      "dataSource": "wikipedia-canned",
      "granularity": "all",
      "dimensions": [
        "language",
        "page"
      ],
      "aggregations": [
        {
          "type": "longSum",
          "fieldName": "count",
          "name": "edit_count_inner_1"
        },
        {
          "type": "doubleSum",
          "fieldName": "added",
          "name": "added_inner_1"
        },
        {
          "type": "javascript",
          "name": "js_inner1",
          "fieldNames": [
            "count"
          ],
          "fnAggregate": "function(current, v) { return current + 2*v; } ",
          "fnCombine": "function(partialA, partialB) { return partialA + partialB; }",
          "fnReset": "function() { return 0; }"
        }
      ],
      "postAggregations": [
        {
          "type": "arithmetic",
          "name": "post_agg",
          "fn": "+",
          "fields": [
            {
              "type": "constant",
              "name": "myConstant",
              "value": 1
            },
            {
              "type": "fieldAccess",
              "name": "edit_count_inner_1",
              "fieldName": "edit_count_inner_1"
            }
          ]
        }
      ],
      "intervals": [
        "2012-01-01T00:00:00.000\/2020-01-03T00:00:00.000"
      ]
    }
  },
  "granularity": "hour",
  "dimensions": [
    "language"
  ],
  "aggregations": [
    {
      "type": "longSum",
      "fieldName": "edit_count_inner_1",
      "name": "edit_count_outer_1"
    },
    {
      "type": "longSum",
      "fieldName": "js_inner1",
      "name": "edit_count_outer_2"
    },
    {
      "type": "longSum",
      "fieldName": "added_inner_1",
      "name": "added_outer_1"
    },
    {
      "type": "longSum",
      "fieldName": "post_agg",
      "name": "postAgg_outer"
    },
    {
      "type": "javascript",
      "name": "js_outer_1",
      "fieldNames": [
        "post_agg",
        "edit_count_inner_1"
      ],
      "fnAggregate": "function(current, v,v2) { return current + v+v2; } ",
      "fnCombine": "function(partialA, partialB) { return partialA + partialB; }",
      "fnReset": "function() { return 0; }"
    },
    {
      "type": "javascript",
      "name": "js_outer_2",
      "fieldNames": [
        "post_agg"
      ],
      "fnAggregate": "function(current, v) { return current + 2*v; } ",
      "fnCombine": "function(partialA, partialB) { return partialA + partialB; }",
      "fnReset": "function() { return 0; }"
    }
  ],
  "intervals": [
    "2012-01-01T00:00:00.000\/2020-01-03T00:00:00.000"
  ]
}
```
Results
```
[ {
  "version" : "v1",
  "timestamp" : "2012-01-01T07:00:00.000Z",
  "event" : {
    "edit_count_outer_1" : 7,
    "js_outer_1" : 17.0,
    "edit_count_outer_2" : 14,
    "language" : "en",
    "postAgg_outer" : 10,
    "js_outer_2" : 20.0,
    "added_outer_1" : 2802
  }
}, {
  "version" : "v1",
  "timestamp" : "2012-01-01T07:00:00.000Z",
  "event" : {
    "edit_count_outer_1" : 2,
    "js_outer_1" : 6.0,
    "edit_count_outer_2" : 4,
    "language" : "fr",
    "postAgg_outer" : 4,
    "js_outer_2" : 8.0,
    "added_outer_1" : 683
  }
}, {
  "version" : "v1",
  "timestamp" : "2012-01-01T07:00:00.000Z",
  "event" : {
    "edit_count_outer_1" : 1,
    "js_outer_1" : 3.0,
    "edit_count_outer_2" : 2,
    "language" : "ja",
    "postAgg_outer" : 2,
    "js_outer_2" : 4.0,
    "added_outer_1" : 16
  }
} ]
```